widgets/WallEntry.php

```diff
@@ -87,13 +74,13 @@ public function isInModal()
     }
 
     /**
-     * @inheritdoc
+     * @return string returns the content type specific part of this wall entry (e.g. post content)
      */
-    public function run()
+    protected function renderContent()
     {
         CalendarBaseAssets::register($this->getView());
         /* @var $entry CalendarEntry */
-        $entry = $this->contentObject;
+        $entry = $this->model;
```

Comment:
I'd just overwrite the $model property in the class and overwrite the phpdoc. Not a big issue but doing so you don't have to do such "casting".

Reply:
Not clear what to do, but it seems nothing. I changed the `$this->contentObject` to `$this->model` relating to doc https://github.com/humhub/documentation/blob/master/docs/develop/modules-migrate.md#migration

Reply:
Just a suggestion in order to hint the IDE about the actual model type:

```
class WallEntry extends WallStreamModuleEntryWidget
{
    /**
     * @var CalendarEntry 
     */
    public $model;

    //....
}
```

Reply:
done in commit https://github.com/humhub/humhub-modules-calendar/commit/45b339a206d70f60fab44a91c7013ad5e0dfe204